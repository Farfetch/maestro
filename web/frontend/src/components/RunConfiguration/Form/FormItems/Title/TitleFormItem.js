import { QuestionCircleOutlined } from "@ant-design/icons";
import { Form, Input, Select, Tooltip } from "antd";
import React from "react";

const TitleFormItem = () => (
  <>
    <Form.Item label="Title">
      <div style={{ display: "flex", alignItems: "center" }}>
        <Form.Item
          name="title"
          noStyle
          rules={[
            {
              required: true,
              message: "Please enter Title"
            }
          ]}
          style={{ flex: 1 }}
        >
          <Input />
        </Form.Item>
        <Tooltip
          title={
            <div>
              You can use tokens in curly braces to dynamically replace them
              with actual values. <br />
              <strong>{"{NUM_AGENTS}"}</strong> returns the number of agents.{" "}
              <br />
              <strong>{"{MAX_RPS}"}</strong> returns the maximum RPS value.{" "}
              <br />
              Any <strong>{"{Custom Property Name}"}</strong> will return the
              custom property value.
            </div>
          }
          placement="topRight"
        >
          <QuestionCircleOutlined style={{ marginLeft: "5px" }} />
        </Tooltip>
      </div>
    </Form.Item>
    <Form.Item label="Labels">
      <Form.Item name="labels" noStyle rules={[]}>
        <Select
          mode="tags"
          allowClear
          style={{ width: "100%" }}
          placeholder="Select default labels"
        />
      </Form.Item>
    </Form.Item>
  </>
);

export default TitleFormItem;
