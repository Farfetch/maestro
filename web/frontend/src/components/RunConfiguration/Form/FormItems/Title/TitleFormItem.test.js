import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import TitleFormItem from "./TitleFormItem";

afterEach(cleanup);

describe("components/RunConfiguration/Form/TitleFormItem", () => {
  test(`should render TitleFormItem component`, async () => {
    const { container } = render(
      <Form>
        <TitleFormItem />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
