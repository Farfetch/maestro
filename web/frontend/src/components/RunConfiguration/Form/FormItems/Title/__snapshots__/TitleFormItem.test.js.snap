// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/TitleFormItem should render TitleFormItem component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal\\"><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Title\\">Title</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><input type=\\"text\\" id=\\"title\\" class=\\"ant-input\\" value=\\"\\"></div></div></div></div><div class=\\"ant-row ant-form-item\\"><div class=\\"ant-col ant-form-item-label\\"><label class=\\"\\" title=\\"Labels\\">Labels</label></div><div class=\\"ant-col ant-form-item-control\\"><div class=\\"ant-form-item-control-input\\"><div class=\\"ant-form-item-control-input-content\\"><div class=\\"ant-select ant-select-multiple ant-select-allow-clear ant-select-show-search\\" style=\\"width: 100%;\\"><div class=\\"ant-select-selector\\"><div class=\\"ant-select-selection-overflow\\"><div class=\\"ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix\\" style=\\"opacity: 1;\\"><div class=\\"ant-select-selection-search\\" style=\\"width: 0px;\\"><input id=\\"labels\\" autocomplete=\\"off\\" type=\\"search\\" class=\\"ant-select-selection-search-input\\" role=\\"combobox\\" aria-haspopup=\\"listbox\\" aria-owns=\\"labels_list\\" aria-autocomplete=\\"list\\" aria-controls=\\"labels_list\\" aria-activedescendant=\\"labels_list_0\\" value=\\"\\"><span class=\\"ant-select-selection-search-mirror\\" aria-hidden=\\"true\\">&nbsp;</span></div></div></div><span class=\\"ant-select-selection-placeholder\\">Select default labels</span></div></div></div></div></div></div></form></div>"`;
